### PR TITLE
QueryScalar fix for #13846

### PR DESCRIPTION
**Pull request by alex-code**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #13846

### CODE REVIEW

**Review by samdark — APPROVED** [review]

File: tests/framework/db/QueryTest.php [hunk 1]
```diff
@@ -340,6 +340,9 @@ public function testCount()
         // testing that orderBy() should be ignored here as it does not affect the count anyway.
         $count = (new Query)->from('customer')->orderBy('status')->count('*', $db);
         $this->assertEquals(3, $count);
+        
+        $count = (new Query)->from('customer')->orderBy('id')->limit(1)->count('*', $db);
```
Comment (samdark) [inline tests/framework/db/QueryTest.php]: Is `limit(1)` necessary here?

**Review by samdark — CHANGES_REQUESTED** [review]

File: tests/framework/db/QueryTest.php (hunk 1 above)
Comment (samdark) [inline tests/framework/db/QueryTest.php]: @Alex-Code ?

Comment (Alex-Code) [inline tests/framework/db/QueryTest.php]: `LIMIT` gets ignored on `COUNT` for MySQL so it was to test if a limit was set it still returned 3.
Not sure if this behaviour is different on MSSQL.

**Comment by samdark** [comment]
Merged https://github.com/yiisoft/yii2/commit/7c6620a65f65f8e7dd151d09c1918001e4e46712. Thanks!

**Pull request closed without merging**